pyaerocom/units_helpers.py

```diff
@@ -151,38 +152,69 @@ def convert_unit_back(data, from_unit, to_unit, var_name=None):
     to_unit : cf_units.Unit or str
         new unit of input data
     var_name : str, optional
-        name of variable. If provided, method 
+        name of variable. If provided, method
         :func:`unit_conversion_fac_custom` is called before the standard unit
         conversion is applied. That requires that `var_name` is specified in
         :attr:`pyaerocom.molmasses.CONV_MUL_FACS`.
-    
+
     Returns
     -------
     data
         data in new unit
-    
+
     """
     if var_name in UCONV_MUL_FACS.index:
-        from_unit, pre_conv_fac = unit_conversion_fac_custom(var_name, 
+        from_unit, pre_conv_fac = unit_conversion_fac_custom(var_name,
                                                              to_unit)
         data = np.divide(data, pre_conv_fac)
-    
+
     conv_fac = unit_conversion_fac(to_unit, from_unit)
     if conv_fac != 1:
         data = np.divide(data, conv_fac)
     return data
 
+
+def implicit_to_explicit_rates(gridded, ts_type):
+    """
+    Convert implicitly defined daily, monthly or yearly rates to
+    per second.
+
+    Some data should be per second but have units without time information
+    information.
+
+    Parameters
+    ----------
+    gridded : GriddedData
+        Data to convert
+    ts_type : str
+        Temporal resolution of gridded.
+    """
+    # TODO: Only modify data if conversion is succesful
+    unit = gridded.units
+    unit_string = str(unit)
+    is_rate = ('/s' in unit_string) or ('s-1' in unit_string)
+    if is_rate:
+        return False
+    else:
+        timestamps = gridded.time_stamps()
+        seconds_factor = seconds_in_periods(timestamps, ts_type)
+        for i in range(len(seconds_factor)):
+            # gridded.to_xarray().isel(time=i).values *= 10**-6  # mg -> kg
+            gridded.to_xarray().isel(time=i).values /= seconds_factor[i]
```

Comment:
@ejgal I wonder if this is really modifying the input `gridded` data object (and if this is always the case, i.e. whether the data is lazy or not)? And is the for loop needed?